AmplifyPlugins/Core/AWSPluginsCore/Model/Support/Model+GraphQL.swift

```diff
@@ -44,7 +44,16 @@ extension Model {
                 if case let .belongsTo(_, targetName) = field.association {
                     fieldName = targetName ?? fieldName
                 }
-                input[fieldName] = (value as? Model)?.id
+                if let modelValue = value as? Model {
+                    input[fieldName] = modelValue.id
+
+                } else if let value = value as? [String: JSONValue],
+                   case let .string(primaryKeyValue) = value[modelSchema.primaryKey.name] {
```

Comment:
nit: prefer `let` and `var` bindings for each individual element. Turns out it's not just a clearer style practice, it also avoids some unexpected behavior if your bound variable shadows something in an outer scope: https://google.github.io/swift/#pattern-matching
```suggestion
                   case .string(let primaryKeyValue) = value[modelSchema.primaryKey.name] {
```